saq/job.py

```diff
@@ -233,7 +236,7 @@ async def update(self, **kwargs) -> None:
             setattr(self, k, v)
         await self.get_queue().update(self)
 
-    async def refresh(self, until_complete: t.Optional[float] = None) -> None:
+    async def refresh(self, until_complete: float | None = None) -> None:
```

Comment:
is this syntax legal in 3.7?

Reply:
I was just about to write an explanation.

The `from __future__ import annotations` means that all annotations are strings at runtime, so no runtime errors irrespective of version, and type checkers are able to use the current syntax irrespective of the version.

I've been running the tests locally on 3.8 without issue and that syntax was introduced in 3.10. Similarly, the builtin subscription syntax, e.g., `list[...]` is 3.9+ but the future annotations mean we don't need to worry about the runtime errors.

Reply:
Of course, if there is an issue I'm missing here, or you'd prefer to stay syntactically valid with the min version irrespective of the future annotations, I'll be happy to roll them back. Just that the current syntax is a lot nicer than what was available in 3.7.

Reply:
I think mypy will automatically add an Optional type when the default value is None? I could have that wrong, though.

Reply:
You are correct, but reliance on it is discouraged: https://github.com/python/peps/pull/689 and https://github.com/python/mypy/issues/9091